projects/observability/src/shared/graphql/request/builders/specification/specification-builder.ts

```diff
@@ -50,4 +74,24 @@ export class SpecificationBuilder {
       })
     };
   }
+
+  private buildSpecName(specs: Specification[]): string {
+    return specs.map(specification => specification.resultAlias()).join('_');
+  }
+
+  private extractForSpecificationsObject(
+    serverData: Dictionary<unknown>,
+    specifications: Dictionary<Specification>
+  ): Dictionary<unknown> {
+    const extractedData: Dictionary<unknown> = {};
+    const specs = Object.values(specifications);
+
+    return Object.keys(specifications)
+      .map((key, index) => [key, specs[index].extractFromServerData(serverData)] as [string, unknown])
+      .reduce((acc, val) => {
+        acc[val[0]] = val[1];
+
+        return acc;
+      }, extractedData);
```

Comment:
```suggestion
      .reduce((acc, [key, value]) => {
	      return {
		      ...acc,
		      [key]: value
	      }
      }, extractedData);
```

nit: This makes it a little more readable.